### PR TITLE
IDEX-2503: Check that .vfs and .codenvy files are ignored in Subversion system configuration

### PR DESCRIPTION
**Pull request by vzhukovskii**
@stour @vparfonov @skabashnyuk

### CODE REVIEW

File: plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        return GLOBAL_SUBVERSION_CONFIG_FILE_PATH;
+    }
+
+    protected void ensureExistingSvnConfigFile() throws IOException {
+        if (Files.notExists(GLOBAL_SUBVERSION_CONFIG_FILE_PATH)) {
+            Files.createFile(GLOBAL_SUBVERSION_CONFIG_FILE_PATH);
+        }
+    }
+
+    protected void checkAndUpdateConfigFile() throws IOException {
```
Comment (skabashnyuk) [inline plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java]: what this method is doing?

File: plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+     *         if creation was failed
+     */
+    protected void ensureExistingSvnConfigFile() throws IOException {
+        if (Files.notExists(GLOBAL_SUBVERSION_CONFIG_FILE_PATH)) {
+            Files.createFile(GLOBAL_SUBVERSION_CONFIG_FILE_PATH);
+        }
+    }
+
+    /**
+     * Checks existing config file for global-ignore property filling and concatenate system default values need to proper work of Codenvy.
```
Comment (skabashnyuk) [inline plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java]: Can you describe wich way for Codenvy is proper

File: plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+     * Writes content to file.
+     *
+     * @param content
+     *         file content
+     * @param configFile
+     *         file path
+     * @throws IOException
+     *         if write was failed
+     */
+    protected void updateConfigFile(String content, Path configFile) throws IOException {
```
Comment (skabashnyuk) [inline plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java]: Are you going to reuse this method?

File: plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java (hunk 2 above)
Comment (vzhukovskii) [inline plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java]: Added little descriptions which way is proper for Codenvy works. Here is little list of which directories should be ignored on user's level.

**Pull request closed without merging**